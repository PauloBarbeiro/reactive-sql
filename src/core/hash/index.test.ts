import { hash } from './index'

describe('hash', function () {
    it('should create a basic predictable hash', () => {
        const result = hash("SELECT id FROM test;")

        expect(result).toMatchSnapshot()
        expect(result).toEqual(-1631577509)
    })

    it("should create a complex predictable hash", () => {
        const result = hash(
            "DROP TABLE IF EXISTS test;\n"
            + "CREATE TABLE test (id INTEGER, age INTEGER, name TEXT);"
            + "INSERT INTO test VALUES (1, 18, 'Liu');"
            + "INSERT INTO test VALUES (2, 20, 'Paul');"
            + "SELECT id FROM test;"
            + "SELECT age,name FROM test WHERE id=1"
        )

        expect(result).toMatchSnapshot()
        expect(result).toEqual(-1367929158)
    })
});