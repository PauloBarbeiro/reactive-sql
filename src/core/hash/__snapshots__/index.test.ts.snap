// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hash should create a basic predictable hash 1`] = `-1631577509`;

exports[`hash should create a complex predictable hash 1`] = `-1367929158`;
